source/usage-examples/findOne.txt

```diff
@@ -0,0 +1,51 @@
+===============
+Find a Document
+===============
+
+.. default-domain:: mongodb
+
+You can retrieve a single document in a collection by chaining together
+the ``find()`` and ``first()`` methods on a ``MongoCollection`` object.
+You can pass a query filter to the ``find()`` method to query for and
+return documents that match the filter in the collection. If you do not
+include a filter, MongoDB returns all the documents in the collection. The
+``first()`` method returns the first document from the matches.
+
+For more information on querying MongoDB with the Java driver, see our
+guide (TODO: link) on querying documents.
+
+You can also chain other methods to the ``find()`` method
+such as ``sort()`` (TODO: link) which organizes the matched documents in a
+specified order, and ``projection`` (TODO: link) which configures the
+included fields in the return document.
+
+The ``find()`` returns an instance of an iterator. Iterators are helpers that
+allow you to traverse and retrieve a group of results. The ``first()`` method
```

Comment:
Are we talking about standard java collection iterators here, or are we talking about mongodb driver specific iterators (cursor wrappers, really)? Given the popularity of iterators in java, I'm guessing that we'll have a page that talks about lazy/eager cursor fetching, so we might be able to just link to that doc here.

Might also be helpful to point out if calling ``first()`` has any side-effects -- can I check it for null and then access the item? Just a single sentence clarifying this would probably help a lot of readers, though you may just prefer that they learn from your code sample below.

Reply:
The iterators definition applies to both MongoIterable and iterators in general. I think this brief explanation belongs in a usage example as we want to provide enough context for them to understand the example without having to read longer explanatory guides.

I think this helps explain your question about side-effects on `first()` as it's called on the iterable which is not null. The document that `first()` returns is null as well as other operations that attempt to access the results from the iterable.